neutrino-custom/base.js

```diff
@@ -0,0 +1,189 @@
+'use strict';
+const path = require('path');
+const webpack = require('webpack');
+const reactPreset = require('neutrino-preset-react');
+const HtmlPlugin = require('html-webpack-plugin');
+const Md5HashPlugin = require('webpack-md5-hash');
+
+const CWD = process.cwd();
+const SRC = path.join(CWD, 'src'); // neutrino's default source directory
+const UI = path.join(CWD, 'ui');
+const DIST = path.join(CWD, 'dist');
+const INDEX_TEMPLATE = path.join(UI, 'index.html');
+const ADMIN_TEMPLATE = path.join(UI, 'admin.html');
+const PERF_TEMPLATE = path.join(UI, 'perf.html');
+const LOGVIEWER_TEMPLATE = path.join(UI, 'logviewer.html');
+const FAILUREVIEWER_TEMPLATE = path.join(UI, 'failureviewer.html');
+const USERGUIDE_TEMPLATE = path.join(UI, 'userguide.html');
+
+const HTML_MINIFY_OPTIONS = {
+    useShortDoctype: true,
+    keepClosingSlash: true,
+    collapseWhitespace: true,
+    preserveLineBreaks: true
+};
+
+module.exports = neutrino => {
+    reactPreset(neutrino);
+
+    // Change the ouput path from build/ to dist/:
+    neutrino.config.output.path(DIST);
+
+    if (process.env.NODE_ENV !== 'test') {
+        // Include files from node_modules in the separate, more-cacheable vendor chunk:
```

Comment:
Hmm, yes -- that was my initial approach, but getting it to work with webpack-chain/Neutrino was nonobvious, although @eliperelman may have some ideas :). As I understand it, Neutrino sets up an empty commons chunk plugin and waits for you to add a vendor entrypoint like this that takes care of determining what will go into it. Using this mechanism, I don't think we can just do something like `reqiure('package.json').dependencies' to get a list of the right modules, since these `add(thing)` calls expect to find a javascript file eventually -- this means that for libs like bootstrap, which provide both css and js and have no default export, we're missing the right path for the js export by default (fontawesome was also a problem here because it's not even js).

I think another approach I could try is removing the default commons chunk plugin altogether and setting up another one based on the implicit scheme. I will give it a try today!